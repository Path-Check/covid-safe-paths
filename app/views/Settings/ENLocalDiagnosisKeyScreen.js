import React, { useContext, useEffect, useState } from 'react';
import { BackHandler, FlatList, StyleSheet, View, Text } from 'react-native';

import { NavigationBarWrapper, Typography } from '../../components';
import ExposureNotificationContext from '../../ExposureNotificationContext';

export const ENLocalDiagnosisKeyScreen = ({ navigation }) => {
  const { diagnosisKeys, fetchDiagnosisKeys } = useContext(
    ExposureNotificationContext,
  );
  const [data, setData] = useState(null);
  useEffect(() => {
    if (data == null) {
      fetchDiagnosisKeys();
    }
    const handleBackPress = () => {
      navigation.goBack();
      return true;
    };

    BackHandler.addEventListener('hardwareBackPress', handleBackPress);

    setData(diagnosisKeys);

    return () => {
      BackHandler.removeEventListener('hardwareBackPress', handleBackPress);
    };
  }, [navigation, diagnosisKeys]);

  const backToSettings = () => {
    navigation.goBack();
  };

  return (
    <NavigationBarWrapper
      title={'Local Diagnosis Keys'}
      onBackPress={backToSettings}>
      <FlatList
        data={data}
        keyExtractor={(item) => item.rollingStartNumber}
        renderItem={(item) => (
          <View style={styles.flatlistRowView}>
            <Typography style={styles.item} use={'body3'}>
              <Text>Rolling start number: {item.item.rollingStartNumber}</Text>
            </Typography>
          </View>
        )}
      />
    </NavigationBarWrapper>
  );
};

const styles = StyleSheet.create({
  // eslint-disable-next-line react-native/no-color-literals
  flatlistRowView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingTop: 7,
    paddingBottom: 5,
    borderBottomWidth: 1,
    borderColor: '#999999',
  },
  item: {
    padding: 10,
    maxWidth: '90%',
  },
});
