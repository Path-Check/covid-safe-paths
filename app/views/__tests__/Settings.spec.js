import 'react-native';

import { act, render } from '@testing-library/react-native';
import React from 'react';

import { FlagsContext } from '../../helpers/flags';
import * as languages from '../../locales/languages';
import { SettingsScreen } from '../Settings';

jest.mock('../../helpers/General', () => {
  return {
    GetStoreData: jest.fn().mockResolvedValue('true'),
  };
});

jest.useFakeTimers();

const mockSetFlags = jest.fn();

let BACKUP_LOCALE_LIST;
let BACKUP_LOCALE_NAME;

beforeEach(() => {
  BACKUP_LOCALE_LIST = languages.LOCALE_LIST;
  BACKUP_LOCALE_NAME = languages.LOCALE_NAME;
  languages.LOCALE_LIST = [
    { label: 'English', value: 'en' },
    { label: 'Other', value: 'ot' },
  ];
  languages.LOCALE_NAME = [{ en: 'English', ot: 'Other' }];

  jest.spyOn(languages, 'getUserLocaleOverride').mockResolvedValue(undefined);
  jest.spyOn(languages, 'setUserLocaleOverride').mockResolvedValue(undefined);
});

afterEach(() => {
  languages.LOCALE_LIST = BACKUP_LOCALE_LIST;
  languages.LOCALE_NAME = BACKUP_LOCALE_NAME;
});

it('renders correctly', async () => {
  const mockFlags = { google_import: true };
  const { asJSON } = render(
    <FlagsContext.Provider value={[mockFlags, mockSetFlags]}>
      <SettingsScreen />
    </FlagsContext.Provider>,
  );

  await act(async () => {
    jest.runAllTimers();
  });

  await expect(asJSON()).toMatchSnapshot();
});

it('renders correctly (without google import flag)', async () => {
  const mockFlags = {};
  const { asJSON } = render(
    <FlagsContext.Provider value={[mockFlags, mockSetFlags]}>
      <SettingsScreen />
    </FlagsContext.Provider>,
  );

  await act(async () => {
    jest.runAllTimers();
  });

  await expect(asJSON()).toMatchSnapshot();
});
