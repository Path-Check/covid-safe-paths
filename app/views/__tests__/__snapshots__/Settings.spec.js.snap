// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> renders correctly with google import flag 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#4051DB",
        "flexBasis": 0,
        "flexGrow": 0,
        "flexShrink": 1,
      }
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#F7F8FF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4051DB",
          "borderBottomColor": "#5061E6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        style={
          Object {
            "alignItems": "center",
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "width": 60,
            "zIndex": 1,
          }
        }
      >
        <SvgXml
          style={
            Object {
              "color": "#FFF",
              "height": 18,
              "width": 18,
            }
          }
          xml="
  <svg width=\\"23\\" height=\\"24\\" viewBox=\\"0 0 23 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M11.4854 3L2.22803 12.2573L11.4854 21.5146\\" stroke=\\"currentColor\\" stroke-opacity=\\"0.4\\" stroke-width=\\"3\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" />
  </svg>
"
        />
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFF",
            "fontFamily": "IBMPlexSans-Bold",
            "fontSize": 16,
            "letterSpacing": 1,
            "lineHeight": 34,
            "paddingHorizontal": 20,
            "position": "absolute",
            "textAlign": "center",
            "textTransform": "uppercase",
            "width": "100%",
          }
        }
      >
        Dashboard
      </Text>
    </View>
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "paddingBottom": 0,
              "paddingLeft": "6.25%",
              "paddingRight": "6.25%",
              "paddingTop": 0,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 18,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 18,
              }
            }
          >
            <SvgXml
              size={24}
              style={
                Object {
                  "marginRight": 12,
                  "marginTop": 2.5,
                  "maxHeight": 24,
                  "maxWidth": 24,
                }
              }
              xml="<?xml version=\\"1.0\\"?>
<svg xmlns=\\"http://www.w3.org/2000/svg\\" id=\\"Layer_1\\" enable-background=\\"new 0 0 512.418 512.418\\" height=\\"512px\\" viewBox=\\"0 0 512.418 512.418\\" width=\\"512px\\"><g><path d=\\"m437.335 75.082c-100.1-100.102-262.136-100.118-362.252 0-100.103 100.102-100.118 262.136 0 362.253 100.1 100.102 262.136 100.117 362.252 0 100.103-100.102 100.117-262.136 0-362.253zm-10.706 325.739c-11.968-10.702-24.77-20.173-38.264-28.335 8.919-30.809 14.203-64.712 15.452-99.954h75.309c-3.405 47.503-21.657 92.064-52.497 128.289zm-393.338-128.289h75.309c1.249 35.242 6.533 69.145 15.452 99.954-13.494 8.162-26.296 17.633-38.264 28.335-30.84-36.225-49.091-80.786-52.497-128.289zm52.498-160.936c11.968 10.702 24.77 20.173 38.264 28.335-8.919 30.809-14.203 64.712-15.452 99.954h-75.31c3.406-47.502 21.657-92.063 52.498-128.289zm154.097 31.709c-26.622-1.904-52.291-8.461-76.088-19.278 13.84-35.639 39.354-78.384 76.088-88.977zm0 32.708v63.873h-98.625c1.13-29.812 5.354-58.439 12.379-84.632 27.043 11.822 56.127 18.882 86.246 20.759zm0 96.519v63.873c-30.119 1.877-59.203 8.937-86.246 20.759-7.025-26.193-11.249-54.82-12.379-84.632zm0 96.581v108.254c-36.732-10.593-62.246-53.333-76.088-88.976 23.797-10.817 49.466-17.374 76.088-19.278zm32.646 0c26.622 1.904 52.291 8.461 76.088 19.278-13.841 35.64-39.354 78.383-76.088 88.976zm0-32.708v-63.873h98.625c-1.13 29.812-5.354 58.439-12.379 84.632-27.043-11.822-56.127-18.882-86.246-20.759zm0-96.519v-63.873c30.119-1.877 59.203-8.937 86.246-20.759 7.025 26.193 11.249 54.82 12.379 84.632zm0-96.581v-108.254c36.734 10.593 62.248 53.338 76.088 88.977-23.797 10.816-49.466 17.373-76.088 19.277zm73.32-91.957c20.895 9.15 40.389 21.557 57.864 36.951-8.318 7.334-17.095 13.984-26.26 19.931-8.139-20.152-18.536-39.736-31.604-56.882zm-210.891 56.882c-9.165-5.947-17.941-12.597-26.26-19.931 17.475-15.394 36.969-27.801 57.864-36.951-13.068 17.148-23.465 36.732-31.604 56.882zm.001 295.958c8.138 20.151 18.537 39.736 31.604 56.882-20.895-9.15-40.389-21.557-57.864-36.951 8.318-7.334 17.095-13.984 26.26-19.931zm242.494 0c9.165 5.947 17.942 12.597 26.26 19.93-17.475 15.394-36.969 27.801-57.864 36.951 13.067-17.144 23.465-36.729 31.604-56.881zm26.362-164.302c-1.249-35.242-6.533-69.146-15.452-99.954 13.494-8.162 26.295-17.633 38.264-28.335 30.84 36.225 49.091 80.786 52.497 128.289z\\" data-original=\\"#000000\\" class=\\"active-path\\" data-old_color=\\"#000000\\" fill=\\"#3C4ED8\\"/></g> </svg>"
            />
            <View
              style={
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                English
              </Text>
            </View>
          </View>
          <Modal
            animationType="slide"
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          >
            <View
              style={
                Object {
                  "flex": 2,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                style={
                  Object {
                    "backgroundColor": "#000",
                    "flex": 1,
                    "opacity": 0.2,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#FFF",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
              >
                <View>
                  <Text
                    style={
                      Object {
                        "color": "#007AFF",
                        "fontFamily": "IBMPlexSans",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "lineHeight": 24,
                        "marginRight": 22,
                        "marginTop": 16,
                        "textAlign": "right",
                        "writingDirection": "ltr",
                      }
                    }
                    use="body1"
                  >
                    Done
                  </Text>
                </View>
                <View>
                  <View>
                    <RCTPicker
                      items={
                        Array [
                          Object {
                            "label": "English",
                            "textColor": undefined,
                            "value": "en",
                          },
                          Object {
                            "label": "Other",
                            "textColor": undefined,
                            "value": "ot",
                          },
                        ]
                      }
                      selectedIndex={0}
                      style={
                        Array [
                          Object {
                            "height": 216,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </Modal>
        </View>
        <View
          style={
            Object {
              "height": 1,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "marginBottom": "2%",
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": "2%",
            }
          }
        />
        <View
          style={
            Object {
              "height": 1,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "paddingBottom": 0,
              "paddingLeft": "6.25%",
              "paddingRight": "6.25%",
              "paddingTop": 0,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 18,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                Your Healthcare Authority
              </Text>
              <Text
                secondary={true}
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "marginTop": 4,
                    "writingDirection": "ltr",
                  }
                }
                use="body3"
              >
                Select a Healthcare Authority to learn about COVID-19 in your area, where available
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "height": 1,
                "width": "100%",
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 18,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                Latest News
              </Text>
              <Text
                secondary={true}
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "marginTop": 4,
                    "writingDirection": "ltr",
                  }
                }
                use="body3"
              >
                Find the latest updates from Safe Paths and your local Healthcare Authorities, where available
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 1,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "marginBottom": "2%",
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": "2%",
            }
          }
        />
        <View
          style={
            Object {
              "height": 1,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "paddingBottom": 0,
              "paddingLeft": "6.25%",
              "paddingRight": "6.25%",
              "paddingTop": 0,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 18,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                About
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "height": 1,
                "width": "100%",
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 18,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                Legal
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "height": 1,
                "width": "100%",
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 18,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                Feature Flags (Dev mode only)
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 1,
              "width": "100%",
            }
          }
        />
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;

exports[`<Settings /> renders correctly without google import flag 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#4051DB",
        "flexBasis": 0,
        "flexGrow": 0,
        "flexShrink": 1,
      }
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#F7F8FF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4051DB",
          "borderBottomColor": "#5061E6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        style={
          Object {
            "alignItems": "center",
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "width": 60,
            "zIndex": 1,
          }
        }
      >
        <SvgXml
          style={
            Object {
              "color": "#FFF",
              "height": 18,
              "width": 18,
            }
          }
          xml="
  <svg width=\\"23\\" height=\\"24\\" viewBox=\\"0 0 23 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M11.4854 3L2.22803 12.2573L11.4854 21.5146\\" stroke=\\"currentColor\\" stroke-opacity=\\"0.4\\" stroke-width=\\"3\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" />
  </svg>
"
        />
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFF",
            "fontFamily": "IBMPlexSans-Bold",
            "fontSize": 16,
            "letterSpacing": 1,
            "lineHeight": 34,
            "paddingHorizontal": 20,
            "position": "absolute",
            "textAlign": "center",
            "textTransform": "uppercase",
            "width": "100%",
          }
        }
      >
        Dashboard
      </Text>
    </View>
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "paddingBottom": 0,
              "paddingLeft": "6.25%",
              "paddingRight": "6.25%",
              "paddingTop": 0,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 18,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 18,
              }
            }
          >
            <SvgXml
              size={24}
              style={
                Object {
                  "marginRight": 12,
                  "marginTop": 2.5,
                  "maxHeight": 24,
                  "maxWidth": 24,
                }
              }
              xml="<?xml version=\\"1.0\\"?>
<svg xmlns=\\"http://www.w3.org/2000/svg\\" id=\\"Layer_1\\" enable-background=\\"new 0 0 512.418 512.418\\" height=\\"512px\\" viewBox=\\"0 0 512.418 512.418\\" width=\\"512px\\"><g><path d=\\"m437.335 75.082c-100.1-100.102-262.136-100.118-362.252 0-100.103 100.102-100.118 262.136 0 362.253 100.1 100.102 262.136 100.117 362.252 0 100.103-100.102 100.117-262.136 0-362.253zm-10.706 325.739c-11.968-10.702-24.77-20.173-38.264-28.335 8.919-30.809 14.203-64.712 15.452-99.954h75.309c-3.405 47.503-21.657 92.064-52.497 128.289zm-393.338-128.289h75.309c1.249 35.242 6.533 69.145 15.452 99.954-13.494 8.162-26.296 17.633-38.264 28.335-30.84-36.225-49.091-80.786-52.497-128.289zm52.498-160.936c11.968 10.702 24.77 20.173 38.264 28.335-8.919 30.809-14.203 64.712-15.452 99.954h-75.31c3.406-47.502 21.657-92.063 52.498-128.289zm154.097 31.709c-26.622-1.904-52.291-8.461-76.088-19.278 13.84-35.639 39.354-78.384 76.088-88.977zm0 32.708v63.873h-98.625c1.13-29.812 5.354-58.439 12.379-84.632 27.043 11.822 56.127 18.882 86.246 20.759zm0 96.519v63.873c-30.119 1.877-59.203 8.937-86.246 20.759-7.025-26.193-11.249-54.82-12.379-84.632zm0 96.581v108.254c-36.732-10.593-62.246-53.333-76.088-88.976 23.797-10.817 49.466-17.374 76.088-19.278zm32.646 0c26.622 1.904 52.291 8.461 76.088 19.278-13.841 35.64-39.354 78.383-76.088 88.976zm0-32.708v-63.873h98.625c-1.13 29.812-5.354 58.439-12.379 84.632-27.043-11.822-56.127-18.882-86.246-20.759zm0-96.519v-63.873c30.119-1.877 59.203-8.937 86.246-20.759 7.025 26.193 11.249 54.82 12.379 84.632zm0-96.581v-108.254c36.734 10.593 62.248 53.338 76.088 88.977-23.797 10.816-49.466 17.373-76.088 19.277zm73.32-91.957c20.895 9.15 40.389 21.557 57.864 36.951-8.318 7.334-17.095 13.984-26.26 19.931-8.139-20.152-18.536-39.736-31.604-56.882zm-210.891 56.882c-9.165-5.947-17.941-12.597-26.26-19.931 17.475-15.394 36.969-27.801 57.864-36.951-13.068 17.148-23.465 36.732-31.604 56.882zm.001 295.958c8.138 20.151 18.537 39.736 31.604 56.882-20.895-9.15-40.389-21.557-57.864-36.951 8.318-7.334 17.095-13.984 26.26-19.931zm242.494 0c9.165 5.947 17.942 12.597 26.26 19.93-17.475 15.394-36.969 27.801-57.864 36.951 13.067-17.144 23.465-36.729 31.604-56.881zm26.362-164.302c-1.249-35.242-6.533-69.146-15.452-99.954 13.494-8.162 26.295-17.633 38.264-28.335 30.84 36.225 49.091 80.786 52.497 128.289z\\" data-original=\\"#000000\\" class=\\"active-path\\" data-old_color=\\"#000000\\" fill=\\"#3C4ED8\\"/></g> </svg>"
            />
            <View
              style={
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                English
              </Text>
            </View>
          </View>
          <Modal
            animationType="slide"
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          >
            <View
              style={
                Object {
                  "flex": 2,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                style={
                  Object {
                    "backgroundColor": "#000",
                    "flex": 1,
                    "opacity": 0.2,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#FFF",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
              >
                <View>
                  <Text
                    style={
                      Object {
                        "color": "#007AFF",
                        "fontFamily": "IBMPlexSans",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "lineHeight": 24,
                        "marginRight": 22,
                        "marginTop": 16,
                        "textAlign": "right",
                        "writingDirection": "ltr",
                      }
                    }
                    use="body1"
                  >
                    Done
                  </Text>
                </View>
                <View>
                  <View>
                    <RCTPicker
                      items={
                        Array [
                          Object {
                            "label": "English",
                            "textColor": undefined,
                            "value": "en",
                          },
                          Object {
                            "label": "Other",
                            "textColor": undefined,
                            "value": "ot",
                          },
                        ]
                      }
                      selectedIndex={0}
                      style={
                        Array [
                          Object {
                            "height": 216,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </Modal>
        </View>
        <View
          style={
            Object {
              "height": 1,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "marginBottom": "2%",
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": "2%",
            }
          }
        />
        <View
          style={
            Object {
              "height": 1,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "paddingBottom": 0,
              "paddingLeft": "6.25%",
              "paddingRight": "6.25%",
              "paddingTop": 0,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 18,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                Your Healthcare Authority
              </Text>
              <Text
                secondary={true}
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "marginTop": 4,
                    "writingDirection": "ltr",
                  }
                }
                use="body3"
              >
                Select a Healthcare Authority to learn about COVID-19 in your area, where available
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "height": 1,
                "width": "100%",
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 18,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                Latest News
              </Text>
              <Text
                secondary={true}
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "marginTop": 4,
                    "writingDirection": "ltr",
                  }
                }
                use="body3"
              >
                Find the latest updates from Safe Paths and your local Healthcare Authorities, where available
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 1,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "marginBottom": "2%",
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": "2%",
            }
          }
        />
        <View
          style={
            Object {
              "height": 1,
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
              "paddingBottom": 0,
              "paddingLeft": "6.25%",
              "paddingRight": "6.25%",
              "paddingTop": 0,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 18,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                About
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "height": 1,
                "width": "100%",
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 18,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                Legal
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "height": 1,
                "width": "100%",
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 18,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 18,
                    "fontWeight": "normal",
                    "lineHeight": 24,
                    "writingDirection": "ltr",
                  }
                }
                use="body1"
              >
                Feature Flags (Dev mode only)
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 1,
              "width": "100%",
            }
          }
        />
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
